design/anova/adding_anova_to_mmrm_design_document.qmd

```diff
@@ -0,0 +1,157 @@
+
+## Additional considerations
+
+### Providing only one fit to `anova.mmrm()`
+
+`?stats::anova` asserts that if only one model fit is provided, the function will calculate the significance of model terms. The `anova.mmrm()` should follow this lead and provide these results accordingly.
+
+### Following the lead of `anova()` methods in `nlme` and `lme4`
+
+In order that `anova.mmrm()` may be comparable to `anova()` methods in the `nlme` and `lme4` packages, it shall contain the following arguments:
+
+-   `test` argument to opt in/out of obtaining LRT results (see `?nlme::anova.lme`)
+-   `refit` arguments to allow for refitting the models (see `?lme4::anova.merMod`)
+
+# Prerequisite: augment `logLik.mmrm_tmb()`
+
+This function needs to return more data in the `attributes()` of the returned value:
+
+-   `nall`, `nobs`, and `df`
+-   Add `"logLik"` to the `class()` of the returned value
+
+# Designing `anova.mmrm()`
+
+## Function specification
+
+### Usage
+
+```{r eval=FALSE}
+## S3 method for class 'mmrm'
+anova(object, ..., test = TRUE, refit = FALSE)
+```
+
+### Arguments
+
+|  |  |
+|----|----|
+| `object` | An object inheriting from class `mmrm`: a fitted mixed model with repeated measures. |
+| `...` | Other optional fitted model objects inheriting from class `mmrm`. |
+| `test` | Logical indicating whether likelihood ratio tests should be used to compare the fitted models represented by `object` and the object(s) in `...`. Defaults to `TRUE`. |
+| `refit` | Logical indicating if the models should be refitted before comparing. Defaults to `FALSE`. |
+
+## Function outline
+
+### Plain language
+
+#### Special case: supplying only one model fit
+
+If the user provides only one model fit to `anova.mmrm()`, the function will calculate and return the significance of the model terms.
+
+#### Comparing multiple model fits
+
+If the user provides more than one model fit, a `tibble` of results is produced. This `tibble` will have a row for each model, and the columns will contain identifying information as well as standard diagnostics (e.g., AIC, log likelihood, etc.).
+
+If the user supplied `test = TRUE`, the model fits are subject to further tests to ensure they are nestable. If the data sets for the fits are not the same and the user supplied `refit = TRUE`, the data sets are harmonized and the models are refitted using the harmonized data set. The pairs of models then undergo likelihood ratio testing, and the results are augmented into the `tibble` of results.
+
+### Pseudocode
+
+```{r eval=FALSE}
+anova.mmrm <- function (object, ..., test = TRUE, refit = FALSE) {
+  
+  # If there are objects supplied to "..."
+  if (...length()) {
+    
+    # Gather all objects into one list
+    fits <- list(object, ...)
+    
+    # Ensure all the objects are mmrm fits
+    validate_mmrm_objects(fits)
```

Comment:
side remark: use existing helper and assertion functions as much as possible, avoid creating too many new ones if not needed